include/pacemaker.h

```diff
@@ -0,0 +1,137 @@
+ *
+ * \param[in,out] xml       The destination for the result, as an XML tree.
+ * \param[in]     st        A connection to the STONITH API.
+ * \param[in]     st_opts   Bitfield of stonith_call_options.
+ * \param[in]     target    The node to get history for.
+ * \param[in]     timeout   How long to wait for the operation to complete (in seconds).
+ * \param[in]     quiet     Suppress most output.
+ * \param[in]     verbose   Include additional output.
+ * \param[in]     broadcast Gather fencing history from all nodes.
+ * \param[in]     cleanup   Clean up fencing history after listing.
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_history(xmlNodePtr *xml, stonith_t *st, unsigned int st_opts,
+                         char *target, int timeout, bool quiet, int verbose,
+                         bool broadcast, bool cleanup);
+
+/*!
+ * \brief List all installed STONITH agents.
+ *
+ * \note If the xml parameter is not NULL, it will be freed first and the previous
+ *       contents lost.
+ *
+ * \param[in,out] xml     The destination for the result, as an XML tree.
+ * \param[in]     st      A connection to the STONITH API.
+ * \param[in]     timeout How long to wait for the operation to complete (in seconds).
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_installed(xmlNodePtr *xml, stonith_t *st, int timeout);
+
+/*!
+ * \brief When was a device last fenced?
+ *
+ * \note If the xml parameter is not NULL, it will be freed first and the previous
+ *       contents lost.
+ *
+ * \param[in,out] xml       The destination for the result, as an XML tree.
+ * \param[in]     target    The target of the fencing operation.
+ * \param[in]     as_nodeid 
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_last_fenced(xmlNodePtr *xml, const char *target, bool as_nodeid);
+
+int pcmk_stonith_list_targets(xmlNodePtr *xml, stonith_t *st, unsigned int st_opts,
+                              char *target, int timeout);
+
+/*!
+ * \brief Get metadata for a resource.
+ *
+ * \note If the xml parameter is not NULL, it will be freed first and the previous
+ *       contents lost.
+ *
+ * \param[in,out] xml     The destination for the result, as an XML tree.
+ * \param[in]     st      A connection to the STONITH API.
+ * \param[in]     agent   The STONITH agent to get metadata for.
+ * \param[in]     timeout How long to wait for the operation to complete (in seconds).
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_stonith_metadata(xmlNodePtr *xml, stonith_t *st, char *agent, int timeout);
+
+int pcmk_stonith_registered(xmlNodePtr *xml, stonith_t *st, unsigned int st_opts,
+                            char *target, int timeout);
+
+int pcmk_stonith_register_level(stonith_t *st, unsigned int st_opts, char *target,
+                                int fence_level, stonith_key_value_t *devices);
```

Comment:
FYI, when we eventually refactor stonith_t, I plan on getting rid of stonith_key_value_t and using a list of pcmk_nvpair_t instead. We can deal with that when the refactor happens.